gobblin-utility/src/main/java/gobblin/util/callbacks/CallbacksDispatcher.java

```diff
@@ -43,6 +45,7 @@
 public class CallbacksDispatcher<L> {
   private final Logger _log;
   private final List<L> _listeners = new ArrayList<>();
+  private final List<WeakReference<L>> _autoListeners = new ArrayList<>();
```

Comment:
I would prefer a `WeakHashMap` here, which manages itself and you don't have to create weak references.